src/django/api/serializers/submission.py

```diff
@@ -0,0 +1,22 @@
+from rest_framework.serializers import ModelSerializer, CharField
+
+from .route_based_serializer_mixin import RouteBasedSerializerMixin
```

Comment:
It's unclear what this does

Reply:
Ah is this for having different fields when pulling the List view vs Detail view? In this case, prefer having two separate serializer classes for the List and Detail views, and using `get_serializer_class()` in `SubmissionsViewSet` to pick the right one dynamically. That would clearer than the metaprogramming the in mixin.

Reply:
Example: https://stackoverflow.com/a/22755648/6995854

Reply:
Cool. I was looking for a way to do exactly that, but could not find it in the documentation.